apache-hadoop-mixin/dashboards/hadoop-namenode-overview.libsonnet

```diff
@@ -0,0 +1,834 @@
+          tooltip: false,
+          viz: false,
+        },
+        lineInterpolation: 'linear',
+        lineWidth: 1,
+        pointSize: 5,
+        scaleDistribution: {
+          type: 'linear',
+        },
+        showPoints: 'auto',
+        spanNulls: false,
+        stacking: {
+          group: 'A',
+          mode: 'none',
+        },
+        thresholdsStyle: {
+          mode: 'off',
+        },
+      },
+      mappings: [],
+      thresholds: {
+        mode: 'absolute',
+        steps: [
+          {
+            color: 'green',
+            value: null,
+          },
+        ],
+      },
+      unit: '',
+    },
+    overrides: [],
+  },
+  options: {
+    legend: {
+      calcs: [],
+      displayMode: 'list',
+      placement: 'bottom',
+      showLegend: true,
+    },
+    tooltip: {
+      mode: 'multi',
+      sort: 'desc',
+    },
+  },
+};
+
+local transactionsSinceLastCheckpointPanel = {
+  datasource: promDatasource,
+  targets: [
+    prometheus.target(
+      'hadoop_namenode_transactionssincelastcheckpoint{job=~"$job", instance=~"$instance", hadoop_cluster=~"$hadoop_cluster", name="FSNamesystem"}',
+      datasource=promDatasource,
+      legendFormat='{{hadoop_cluster}} - {{instance}}',
+      format='time_series',
+    ),
+  ],
+  type: 'timeseries',
+  title: 'Transactions since last checkpoint',
+  description: 'Number of transactions processed by the NameNode since the last checkpoint.',
+  fieldConfig: {
+    defaults: {
+      color: {
+        mode: 'palette-classic',
+      },
+      custom: {
+        axisCenteredZero: false,
+        axisColorMode: 'text',
+        axisLabel: '',
+        axisPlacement: 'auto',
+        barAlignment: 0,
+        drawStyle: 'line',
+        fillOpacity: 0,
+        gradientMode: 'none',
+        hideFrom: {
+          legend: false,
+          tooltip: false,
+          viz: false,
+        },
+        lineInterpolation: 'linear',
+        lineWidth: 1,
+        pointSize: 5,
+        scaleDistribution: {
+          type: 'linear',
+        },
+        showPoints: 'auto',
+        spanNulls: false,
+        stacking: {
+          group: 'A',
+          mode: 'none',
+        },
+        thresholdsStyle: {
+          mode: 'off',
+        },
+      },
+      mappings: [],
+      thresholds: {
+        mode: 'absolute',
+        steps: [
+          {
+            color: 'green',
+            value: null,
+          },
+        ],
+      },
+      unit: '',
+    },
+    overrides: [],
+  },
+  options: {
+    legend: {
+      calcs: [],
+      displayMode: 'list',
+      placement: 'bottom',
+      showLegend: true,
+    },
+    tooltip: {
+      mode: 'multi',
+      sort: 'desc',
+    },
+  },
+};
+
+local volumeFailuresPanel = {
+  datasource: promDatasource,
+  targets: [
+    prometheus.target(
+      'increase(hadoop_namenode_volumefailurestotal{job=~"$job", instance=~"$instance", hadoop_cluster=~"$hadoop_cluster", name="FSNamesystem"}[$__interval:])',
```

Comment:
general question, no fix, but why was `name="FSNamesystem`  filter added on?

Reply:
So the NameNode is responsible for managing the FS across many DataNodes and the metrics produce a name with FSNamesystem and FSNamesystemState which appear to be the same value while testing. The FSNamesystem provides a general state of volume failures across all your datanodes in the Hadoop filesystem. The FSNamesystem provides the current state of the Namenode along with volume failures. From experimentation, these values look to be producing the same metric even while load is applied like in this this chart:
![Screenshot 2023-07-13 at 2 53 51 PM](https://github.com/grafana/jsonnet-libs/assets/13648435/3b03ef29-230c-439d-9249-2180ea6d9ac7)

I don't believe there is any added value including both.